server/src/main/java/au/org/aodn/ogcapi/server/core/service/ElasticSearch.java

```diff
@@ -215,74 +221,153 @@ protected List<Hit<CategorySuggestDTO>> getCategorySuggestions(String input) thr
 
         return new ResponseEntity<>(allSuggestions, HttpStatus.OK);
     }
+    /**
+     * Core function to do the search, it used pageableSearch to make sure all records are return.
+     *
+     * @param queries
+     * @param should
+     * @param filters
+     * @param properties
+     * @param maxSize: Max number of records to be return
+     * @return
+     * @throws IOException
+     */
+    protected List<StacCollectionModel> searchCollectionBy(final List<Query> queries,
+                                                           final List<Query> should,
+                                                           final List<Query> filters,
+                                                           final List<String> properties,
+                                                           final Long maxSize) {
+
+        Supplier<SearchRequest.Builder> builderSupplier = () -> {
+            SearchRequest.Builder builder = new SearchRequest.Builder();
+            builder.index(indexName)
+                    .size(pageSize)
+                    .query(q -> q.bool(createBoolQueryForProperties(queries, should, filters)))
+                    .sort(so -> so.score(v -> v.order(SortOrder.Desc)))
+                    .sort(so -> so
+                            .field(FieldSort.of(f -> f
+                                    .field(StacSummeries.Score.searchField)
+                                    .order(SortOrder.Desc))))
+                    .sort(so -> so
+                            // We need a unique key for the search, so _id is the best
+                            .field(FieldSort.of(f -> f
+                                    .field("id.keyword")
```

Comment:
I'm not clear here, the id values are UUID random code rather than incremental digits or ordered alphabet characters, how can it help by sorting them either ASC or DESC?

Reply:
The sort is based on the two above criteria first, then if they have both order, then last order is id, the way elastic recommend is you keep a fix sorting order for a particular query that can have record later then 2K, then the 2001 record is obtained by using the same sorting order query plus the search_after, in this case the id. 

The two initial criteria is not unique enough as they can have the same mathematical value, hence you need something unique.